crates/storage/src/blueprint/sparse.rs

```diff
@@ -152,7 +155,7 @@ where
                 storage.storage::<Metadata>().remove(primary_key)?;
             } else {
                 // Generate new metadata for the updated tree
-                let metadata = SparseMerkleMetadata { root };
+                let metadata = SparseMerkleMetadataV1 { root }.into();
```

Comment:
Maybe it is okay to add a new constructor that creates `SparseMerkleMetadata` from the root.